File: 238-product-of-array-except-self/product-of-array-except-self.ts
function productExceptSelf(nums: number[]): number[] {
    let n = nums.length
    let pre =  Array(n).fill(1);
    for (let i = 1;i<n;i++){
        pre[i] = pre[i-1]*nums[i-1]
    }
    let post = 1;
    let ret = Array(n).fill(0)
    for (let i = n-1; i>=0;i--){
        ret[i] = post*pre[i]
        post*=nums[i]
    }
    return ret;  
};

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        pre = [1]*(n)
        for i in range(1,n):
            pre[i] = nums[i-1]*pre[i-1]
        print(pre)
        post = 1
        ret = [0]*n
        print(ret)
        for i in range(n-1,-1,-1):
            ret[i] = pre[i]*post
            post*=nums[i]
            
        return ret